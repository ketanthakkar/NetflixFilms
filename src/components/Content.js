// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import Filter from './Filter';
import MovieItem from './MovieItem';
import ErrorBoundary from '../error/ErrorBoundary';
import Header from './Header';
import Footer from './Footer';
import NoMovieFound from './NoMovieFound';
import { fetchSearchedMovies } from '../actions/index';

type Props = {
  movies: [],
  match: {
    params: {
      query: string
    }
  },
  showSearch: boolean,
  fetchSearchedMovies: Function,
}

class Content extends React.Component<Props> {
  static defaultProps = {
    showSearch: true,
  };
  
  static fetchData(dispatch, match) {
    return dispatch(fetchSearchedMovies(match.params.query));
  }

  componentDidMount() {
    if (this.props.match) {
      this.props.fetchSearchedMovies(this.props.match.params.query);
    }
  }

  render() {
    const { movies, showSearch } = this.props;
    return (
            <React.Fragment>
                {showSearch && <Header /> }
                <ErrorBoundary>
                    { movies.length > 0 ? <React.Fragment>
                            <Filter />
                            <main className="moviedata-container"> {
                                movies.map(movie => <MovieItem key={movie.id} movieItem={movie} />)
                            }
                            </main>
                        </React.Fragment>
                        : <NoMovieFound /> 
                    }
                </ErrorBoundary>
                {showSearch && <Footer /> }
            </React.Fragment>
    );
  }
}

const mapDispatchToProps = { fetchSearchedMovies };

const mapStateToProps = state => ({
  movies: state.movies.movieData,
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Content));
