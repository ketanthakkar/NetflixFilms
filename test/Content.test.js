import React from 'react';
import { shallow } from 'enzyme';
import Content from '../src/components/Content';

const movies = {
  data: [
    {
      id: 337167,
      title: 'Fifty Shades Freed',
      tagline: "Don't miss the climax",
      vote_average: 6.1,
      vote_count: 1195,
      release_date: '2018-02-07',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      overview: 'Believing they have left behind shadowy figures from their past',
      budget: 55000000,
      revenue: 136906000,
      genres: [
        'Drama',
        'Romance',
      ],
      runtime: 106,
    },
  ],
};

describe('should render Content component', () => {
  const tree = shallow(
    <Content movies={movies.data} />,
  );
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });
});
