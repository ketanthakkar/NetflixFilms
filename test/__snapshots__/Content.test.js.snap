// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Content component Snapshot test 1`] = `
<Route>
  <Component />
</Route>
`;
